client/internal/receivedprocessor.go

```diff
@@ -281,3 +402,16 @@ func (r *receivedProcessor) rcvCommand(ctx context.Context, command *protobufs.S
 		r.callbacks.OnCommand(ctx, command)
 	}
 }
+
+// updateStoredConnectionSettingsStatus returns a bool of if status should replace oldStatus.
+// It's true if:
+// - no oldStatus
+// - hash changes
+// - status changes from APPLYING or UNSET
```

Comment:
I am not sure I understand why this particular set of conditions was chosen. For example if the previous Status for FAILED and the current status is APPLYING or APPLIED, why shouldn't we update it?

Reply:
To reword a bit:
We will always update if there is no old status
We will always update if the old status, and new status have a different hash

If the hash between the old/new status' are the same, we will allow updates from
- `UNSET|APPLYING -> APPLIED|FAILED`
- `ANY -> FAILED`

The reason why `FAILED` if preferred over `APPLIED` is because we have two seperate callbacks that can execute independently; if the server offers connection settings for OpAMP, and something like traces we want to signal a failure if either has an error.